inference-engine/src/cldnn_engine/ops/loop.cpp

```diff
@@ -0,0 +1,255 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#include "cldnn_program.h"
+#include "cldnn_common_utils.h"
+#include "cldnn_engine.h"
+
+#include <cpp/ie_cnn_network.h>
+
+#include "ngraph/op/loop.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/util/sub_graph_base.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include "cldnn/primitives/loop.hpp"
+#include "cldnn/primitives/mutable_data.hpp"
+#include "cldnn/primitives/data.hpp"
+#include "cldnn/primitives/reorder.hpp"
+#include "cldnn/graph/topology.hpp"
+
+#include <vector>
+#include <algorithm>
+
+using Loop = ngraph::op::v5::Loop;
+
+namespace CLDNNPlugin {
+
+template<class DATA_TYPE>
+static DATA_TYPE CreateScalarData(Program &p, const cldnn::primitive_id& id, int64_t num) {
+    auto mem = p.GetEngine().allocate_memory({ cldnn::data_types::i64, cldnn::format::bfyx, { 1, 1, 1, 1 } });
+    cldnn::mem_lock<int64_t> ptr{mem, p.GetEngine().get_program_stream()};
+    *ptr.begin() = num;
+    return {id, mem};
+}
+
+static cldnn::mutable_data CreateAdditionalOutputData(Program &p, const std::shared_ptr<ngraph::Node>& op,
+                                            const cldnn::primitive_id& id, const cldnn::primitive_id& input,
```

Comment:
nit: alignment

Reply:
hello, thanks for the review but this PR is draft to check whether it passes CI. We are going to update this PR to be ready. Thanks!

Reply:
See, sorry!

Reply:
sorry from us! We'll mark more explicitly..

Reply:
Fixed it. thanks.